edx/analytics/tasks/util/record.py

```
@@ -529,6 +579,44 @@ def deserialize_from_string(self, string_value):
         return datetime.date(*[int(x) for x in string_value.split('-')])
 
 
+class DateTimeField(Field):  # pylint: disable=abstract-method
```

I haven't fully validated this yet, but I think we want something like this:

```python
    hive_type = 'TIMESTAMP'
    sql_base_type = 'DATETIME'
    elasticsearch_type = 'date'
    elasticsearch_format = 'yyyy-MM-dd HH:mm:ss.SSSSSS'
    string_format = '%Y-%m-%d %H:%M:%S.%f'  # hive timestamp format
```

That way we can actually query the table in Hive using all of the features of the TIMESTAMP type.

It also occurs to me that we probably want to ensure that the data is stored in UTC. In the validate function we should probably check for that.

Additionally, I'd prefer `validate()` was much less accommodating than it currently is. It should raise an error if the value is not a native python datetime object. I'd prefer it didn't actually try to intelligently parse stuff. Same goes for serialize and deserialize. They should assume you always want the data to appear in `self.string_format` format when in string format.